app/controllers/recipes_foods_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipesFoodsController < ApplicationController
+  def new
+    @recipe = Recipe.find(params[:recipe_id])
+    unless @recipe.user == current_user
+      flash[:alert] = 'You can only add foods to your own recipes.'
+      redirect_to recipe_path(@recipe)
+    end
+    @recipe_food = RecipeFood.new
+    @foods = current_user.foods
+  end
+
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = RecipeFood.new(recipe_food_params)
+    @recipe_food.recipe = @recipe
+    if @recipe_food.save
+      flash[:notice] = 'Food added to recipe.'
+      redirect_to recipe_path(@recipe)
+    else
+      flash[:alert] = 'Food not added to recipe.'
+      render :new
+    end
+  end

```

Comment:
- [ ] There is a bug 🐛 while trying to create a `recipe_food`. The food item never gets added to the recipe when I click the create  Recipe_Food button. Please take a look at the attached image below and try to fix the problem 🙏 
![image](https://user-images.githubusercontent.com/46839548/208216503-82d079dd-0a37-4c6e-8454-e7366bad486a.png)

Reply:
@codepantha Something moved before the lasts commits.  Now it works.  Done !!

Reply:
@codepantha , regarding the tests, we have not found that point in the requirements. That's why we didn't. Could you check please?

Reply:
Glad you've done it 👏